### PR TITLE
Variogram estimation from multiple fields

### PR DESCRIPTION
**Pull request by MuellerSeb**
With this PR a new option for unstructured variogram estimation is added:
You can now estimate one variogram from multiple fields on the same point-set. This could be useful to estimate a single variogram to describe multiple realizations of the same process.

Example:
- time series of precipitation data -> one variogram for all days needed

In addition, these fields can have missing values, denoted by a new `no_data` argument.

Checklist:
- [x] blackened
- [x] test written
- [x] examples added

### CODE REVIEW

**Comment by MuellerSeb** [comment]
@LSchueler ready for review! :smiling_face_with_three_hearts:

File: examples/03_variogram/03_multi_vario.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+###############################################################################
+# Generate two synthetic fields with an exponential model.
+
+field1 = srf((x, y), seed=19970221)
+field2 = srf((x, y), seed=20011012)
+fields = [field1, field2]
+
+###############################################################################
+# Now we estimate the variograms for both fields individual and then again
```
Comment (LSchueler) [inline examples/03_variogram/03_multi_vario.py:26]: individual => individually

File: examples/03_variogram/03_multi_vario.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+bin_center, gamma2 = gs.vario_estimate_unstructured((x, y), field2, bins)
+bin_center, gamma = gs.vario_estimate_unstructured((x, y), fields, bins)
+
+###############################################################################
+# Now we demonstrate, that the mean variogram from both fields coincides
+# with the joined estimated one.
+
+plt.plot(bin_center, gamma1, label="field 1")
+plt.plot(bin_center, gamma2, label="field 2")
+plt.plot(bin_center, gamma, label="field joint")
```
Comment (LSchueler) [inline examples/03_variogram/03_multi_vario.py:40]: "field joint" => "joined fields" ?

File: gstools/variogram/estimator.pyx [hunk 3]
```diff
@@ -150,18 +150,22 @@ def unstructured(
     cdef int i_max = bin_edges.shape[0] - 1
     cdef int j_max = x.shape[0] - 1
     cdef int k_max = x.shape[0]
+    cdef int f_max = f.shape[0]
 
     cdef vector[double] variogram = vector[double](len(bin_edges)-1, 0.0)
     cdef vector[long] counts = vector[long](len(bin_edges)-1, 0)
-    cdef int i, j, k
+    cdef int i, j, k, m
```
Comment (LSchueler) [inline gstools/variogram/estimator.pyx:157]: You skipped my favorite letter!! :-p

**Review by LSchueler — CHANGES_REQUESTED** [review]
That is actually quite nice syntactic sugar. I think I like it. Although it makes it even more complicated dealing with all the different field shapes internally. But that will hopefully one day change with version 2.0.

Shouldn't this be extended to the structured variogram estimation? - I imagine this could lead to confusions, when someone changes from unstructured to structured fields.

And some minor requests in the code comments.

Cheers!

File: gstools/variogram/estimator.pyx (hunk 3 above)
Comment (MuellerSeb) [reply gstools/variogram/estimator.pyx:157]: See here : https://www.python.org/dev/peps/pep-0008/#names-to-avoid

**Comment by MuellerSeb** [comment]
This was added in the `direct-vario` branch with https://github.com/GeoStat-Framework/GSTools/commit/c044b620857fbe0666d942ad02e3960261db54e6

Closing this PR in favor of the `direct-vario` branch.

**Pull request closed without merging**